docs/features/linux_networking/es2k/readme-linux-networking-with-frr.md

```diff
@@ -0,0 +1,303 @@
+```
+
+#### Configure rules for underlay control packets
+
+Configure rules to send underlay control packets from IDPF netdev to physical port.
+
+Below configuration assumes
+- Underlay IDPF netdev has a VSI value 10
+- First physical port will have a port ID of 0
+
+```
+# Configuration for control packets between physical port 0 to underlay IDPF netdev VSI-10
+ p4rt-ctl add-entry br0 linux_networking_control.handle_rx_from_wire_to_ovs_table "vmeta.common.port_id=0,user_meta.cmeta.bit32_zeros=0,action=linux_networking_control.set_dest(26)"
+
+# Configuration for control packets between underlay IDPF netdev VSI-10 to physical port 0
+ p4rt-ctl add-entry br0 linux_networking_control.handle_tx_from_host_to_ovs_and_ovs_to_wire_table "vmeta.common.vsi=10,user_meta.cmeta.bit32_zeros=0,action=linux_networking_control.set_dest(0)"
+```
+
+#### Underlay configuration
+
+Create a dummy interface which is used for TEP termination and IDPF netdev for underlay connectivity. Use BGP protocol with FRR, for route redistribution.
+
+Below configuration assumes
+- Underlay IDPF netdev has a VSI value 10
+
+```
+p4rt-ctl add-entry br0 linux_networking_control.ecmp_lpm_root_lut "user_meta.cmeta.bit32_zeros=4/255.255.255.255,priority=2,action=linux_networking_control.ecmp_lpm_root_lut_action(0)"
+
+nmcli device set <IDPF netdev for VSI 10> managed no
+ip link add dev TEP0 type dummy
+```
+
+FRR running configuration
+```
+# <Install FRR on ACC and enable BGP protocol>  
+# <FRR VTYSH running configuration>
+
+interface TEP0
+ip address 30.1.1.1/24
+exit
+!
+interface <IDPF netdev for VSI 10>
+ip address 70.1.1.1/24
+exit
+!
+router bgp 65000
+bgp router-id 30.1.1.1
+neighbor 70.1.1.1 remote-as 65000
+!
+address-family ipv4 unicast
+network 30.1.1.0/24
+exit-address-family
+```
+
+Sample link partner FRR VTYSH configuration.
+```
+interface TEP0
+ip address 40.1.1.1/24
+exit
+!
+interface <Link Partner Underlay Port>
+ip address 70.1.1.2/24
+exit
+!
+router bgp 65000
+bgp router-id 40.1.1.1
+neighbor 70.1.1.2 remote-as 65000
+!
+address-family ipv4 unicast
+  network 40.1.1.0/24
+exit-address-family
+```
+
+#### Test the ping scenarios
+
+  - Ping between VM's on the same host
+  - Underlay ping
+  - Remote TEP ping
+  - Overlay ping: Ping between VM's on different hosts
+
+## Limitations <a name="limitations"></a>
+
+Current Linux Networking support for the networking recipe has following limitations:
+- All VLAN interfaces created on top of IDPF netdev, should always be in lowercase format "vlan+vlan_id"
+Ex: vlan1, vlan2, vlan3 ... vlan4094.
+- Set the pipeline before adding br-int port, vxlan0 port, and adding vlan ports to br-int bridge.
+- VxLAN destination port should always be standard port. i.e., 4789. (limitation by p4 parser)
+- We do not support any ofproto rules that would prevent FDB learning on OvS.
+- VLAN Tagged packets are not supported.

```

Comment:
Can we instead say something like current P4 doesn't cover Vlan tagged packets?

not supported gives the impression that IPU may not be capable of it?